ruby/wipe-appwrite-bucket/index.rb

```diff
@@ -0,0 +1,55 @@
+require 'appwrite'
+
+def main(req, res)
+  client = Appwrite::Client.new
+  storage = Appwrite::Storage.new(client)
+
+  if !req.variables['APPWRITE_FUNCTION_ENDPOINT'] or !req.variables['APPWRITE_FUNCTION_API_KEY']
+    return res.json({
+      :success => false,
+      :message => 'Environment variables are not set. Function cannot use Appwrite SDK.',
+    })
+  end
+
+  bucket_id = nil
+  begin
+    payload = JSON.parse(req.payload)
+    bucket_id = (payload['bucketId'] || '').delete(' ')
+  rescue Exception => err
+    return res.json({
+      :success => false,
+      :message => 'Payload is invalid.',
+    })
+  end
+
+  if !bucket_id || bucket_id == ''
+    return res.json({
+      :success => false,
+      :message => 'bucketId was not provided.',
+    })
+  end
+
+  client
+    .set_endpoint(req.variables['APPWRITE_FUNCTION_ENDPOINT'])
+    .set_project(req.variables['APPWRITE_FUNCTION_PROJECTID'])
+    .set_key(req.variables['APPWRITE_FUNCTION_API_KEY'])
+    .set_self_signed(true)
+
+  file_list = nil
+  begin
+    file_list = storage.list_files(bucket_id: bucket_id)
```

Comment:
By default, Appwrite has a default limit of 25 so only up to 25 results will return from this `storage.list_files()` API call. To make sure the entire bucket is wiped, you'll need to run this in a loop until all files are deleted.

Reply:
Fixed.